### PR TITLE
HPCC-17207 Create HTML files for Portal

### PR DESCRIPTION
**Pull request by xwang2713**
The generated HTML files will be in docs/PortalHTML directory.
Create a zip file for Portal HTML.
Also add generating ECLRefernece HTML for ECLIDE documentation which will be used to create CHM file during ECLIDE Jenkins's build. The process is automated in the Jenkins job.

Signed-off-by: Xiaoming Wang <xiaoming.wang@lexisnexi.com> 

@g-pan please review the changes and verify the generated HTML and CHM files

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

File: cmake_modules/docMacros.cmake [hunk 1]
```diff
@@ -99,33 +99,47 @@ MACRO(DOCBOOK_TO_PDF _xsl _file _name)
         ENDIF(MAKE_DOCS)
 ENDMACRO(DOCBOOK_TO_PDF targetname_suffix srcfile outfile targetdir deps_list)
 
-MACRO(DOCBOOK_TO_HTML _xsl_file _xml_file _out_dir)
+MACRO(DOCBOOK_TO_HTML _xsl_file _xml_file _out_dir _html_target _css_path _zip_target)
     IF(MAKE_DOCS)
-       set(_html_zip "doc_generate_html_zip")
-       set(_html_files "_generate_html_files")
+       if("${_target_file}" STREQUAL "")
+           set(_tareget "_generate_html_files")
```
Comment (g-pan) [inline cmake_modules/docMacros.cmake]: Looks like _target is spelled wrong (_tareget) Not certain if it was intentional or if it matters.

**Review by g-pan — COMMENTED** [review]
Considerable changes to all xpointers.
Is there a complete build of all docs (PDF, Eclipse, HTML, CHM) that should also be reviewed?

File: cmake_modules/docMacros.cmake (hunk 1 above)
Comment (xwang2713) [inline cmake_modules/docMacros.cmake]: I corrected it and rebuild docs. Here is the chm file : http://10.240.32.242/data3/ming/ECLReference.chm
Please check again. Thanks

**Comment by g-pan** [comment]
Reviewed these changes and all look good.

**Pull request merged**